### PR TITLE
Add loopback interface

### PR DESCRIPTION
**Pull request by wngr**
Makes API used `examples/netcat_host.rs` work.
Any downsides?

### CODE REVIEW

File: machine/src/iface.rs [hunk 1]
```diff
@@ -121,6 +121,32 @@ impl Write for Iface {
...
+                return Err(err);
+            }
+
+            let mut req = ioctl::ifreq::new(&lo);
+            req.ifr_ifru.ifru_flags |=
+                libc::IFF_UP as i16 | libc::IFF_LOOPBACK as i16 | libc::IFF_RUNNING as i16;
+
+            let res = errno!(ioctl::siocsifflags(fd, &req));
+            let _ = libc::close(fd);
+            res?;
```
Comment (dvc94ch) [inline machine/src/iface.rs:147]: if it works, lgtm

**Comment by dvc94ch** [comment]
Can you add the example to ci?

**Comment by dvc94ch** [comment]
this breaks the if-watch example (as it now returns localhost addresses too). can you find a way to fix the example?

**Review by dvc94ch — APPROVED** [review]

File: examples/if_watch.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ use ipnet::IpNet;
 #[async_std::main]
 async fn main() -> Result<()> {
     let mut watcher = IfWatcher::new().await?;
-    for _ in 0..10 {
+    for _ in 0..12 {
```
Comment (dvc94ch) [inline examples/if_watch.rs:8]: one additional event but you're waiting for two

Comment (wngr) [reply examples/if_watch.rs:8]: the issue is the `netcat_host` example, which works on my machine, but not on ci :confused:.

Comment (wngr) [reply examples/if_watch.rs:8]: Anyway, it's two because it yields both `127.0.0.1` and `::1`.

**Pull request merged**